src/azul/terra.py

```diff
@@ -198,6 +198,28 @@ def type_name(self):
     def qualify_table(self, table_name: str) -> str:
         return '.'.join((self.project, self.bq_name, table_name))
 
+    def contains(self, other: 'SourceSpec') -> bool:
+        """
+        >>> s = TDRSourceSpec.parse('tdr:foo:snapshot/bar:')
+        >>> sc = TDRSourceSpec.parse('tdr:foo:snapshot/bar:22')
+        >>> scp =TDRSourceSpec.parse('tdr:foo:snapshot/bar:22/1')
+        >>> s.contains(sc)
+        True
+        >>> s.contains(scp)
+        True
+        >>> sc.contains(s)
+        False
+        >>> scp.contains(s)
+        False
+        """
+        return (
+            super().contains(other)
```

Comment:
Order the terms so that the cheapest ones come first.

Reply:
If we pull up the `isinstance` check to the super class but move `super().compare()` down to be the last item in ordering, There might be attribute errors if someone accidentally compares two different types of SourceSpecs....